flake.nix

```diff
@@ -27,17 +28,29 @@ rec {
   };
 
   outputs =
-    { self
+    inputs@{ 
+      self
+    , flake-parts
+    , git-hooks-nix
     , nixpkgs
-    , flake-utils
     , released-nixpkgs-unstable
     , released-nixpkgs-stable
     , released-nix-unstable
     , released-nix-stable
     , nix-pills
     }:
-    flake-utils.lib.eachDefaultSystem (system:
-      let
+  flake-parts.lib.mkFlake { inherit inputs; } {
+    flake = {};
+    imports = [
+      inputs.git-hooks-nix.flakeModule
+    ];
+    systems = [
```

Comment:
This can be shortened with: https://github.com/loophp/nix-shell/blob/4a8220125e9537dfa316b49e2b9cf77e977c86af/flake.nix#L20